packages/mermaid/src/diagrams/sankey/sankeyDB.ts

```diff
@@ -0,0 +1,124 @@
+// import { log } from '../../logger.js';
+// import mermaidAPI from '../../mermaidAPI.js';
+import * as configApi from '../../config.js';
+import common from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  getAccDescription,
+  setAccDescription,
+  setDiagramTitle,
+  getDiagramTitle,
+  clear as commonClear,
+} from '../../commonDb.js';
+
+// export const parseDirective = function (statement, context, type) {
+//   mermaidAPI.parseDirective(this, statement, context, type);
+// };
+
+// export const cleanupComments = (text: string): string => {
+//   return text.trimStart().replace(/^\s*%%(?!{)[^\n]+\n?/gm, '');
+// };
+let links: Array<Link> = [];
+let nodes: Array<Node> = [];
+let nodesHash: Record<string, Node> = {};
+
+const clear = function () {
+  links = [];
+  nodes = [];
+  nodesHash = {};
+  commonClear();
+};
+
+type Nullable<T> = T | null;
+
+// interface ILink {
+//   source?: Node;
+//   target?: Node;
+//   value?: number;
+// }
+
+class Link {
+  source: Nullable<Node>;
+  target: Nullable<Node>;
+  value: Nullable<number>;
+  constructor() {
+    this.source = null;
+    this.target = null;
+    this.value = 0;
+  }
+}
+
+/**
+ * Adds a link between two elements on the diagram
+ *
+ * @param source - Node where the link starts
+ * @param target - Node where the link ends
+ * @param value - number, float or integer, describes the amount to be passed
+ */
+// const addLink = ({ source, target, amount }: ILink = {}): Link => {
+const addLink = function (source?: Node, target?: Node, value?: number): Link {
+  const link: Link = new Link();
+
+  // TODO: make attribute setters
+  if (source !== undefined) {
+    link.source = source;
+  }
+  if (target !== undefined) {
+    link.target = target;
+  }
+  if (value !== undefined) {
+    link.value = value;
+  }
```

Comment:
```suggestion
// outside fn
class Link {
  constructor(public source?: Node, public target?: Node, public value = 0) {
  }
}
// this fn

  const link: Link = new Link(source, target, value);
```

Reply:
That probably will be redesigned completely, because that is unclear if I a link is allowed not to have one or several attributes. Probably all of them are required, but I have not come to a conclusion yet

Reply:
Fixed